golang/internal/config/secret.go

```diff
@@ -111,6 +73,56 @@ func IsExpiredKey(fileContent string) (bool, error) {
 	return privateKeyObj.IsExpired(), nil
 }
 
-func InjectSecret(secret string, appConfig *CommonConfiguration) {
-	appConfig.SecretPrivateKey = secret
+func ValidateJwtAndCheckNonceBlacklist(secrets SecretStore, unvalidatedToken string) (*ValidJWT, error) {
+	blacklistedNonce, err := secrets.GetBlacklistedNonce()
+	if err != nil {
+		return nil, err
+	}
+
+	token, err := ValidateAndCreateJWT(unvalidatedToken)
+	if err != nil {
+		return nil, err
+	}
+
+	if blacklistedNonce != "" && token.Nonce == blacklistedNonce {
+		return nil, &NonceBlacklistedError{}
+	}
+
+	return token, nil
+}
+
+func (c *CommonConfiguration) InjectPrivateKey(secrets SecretStore) error {
+	key, err := secrets.LoadPrivateKey()
+	if err != nil {
+		return err
+	}
+
+	c.SecretPrivateKey = key
+	return nil
+}
+
+func (c *CommonConfiguration) InjectGrpcToken(secrets SecretStore) error {
+	token, err := secrets.GetConnectionToken()
+	if err == nil && token != "" {
+		c.GrpcToken, err = ValidateJwtAndCheckNonceBlacklist(secrets, token)
+		if err == nil {
+			return nil
+		}
+	}
+
+	token = c.InstallToken
+	if err != nil {
+		log.Error().Err(err).Msg("Failed to validate the connection token. Failing back to the install token")
```

Comment:
@nandor-magyar always tells me that we are using sentences with small capital starters

Reply:
That is true for errors, that is a string.